### DISCUSSION TITLE
Add generated files to a different path

### DISCUSSION THREAD

**Discussion by amoghyermalkar123** [a-h/templ#922] [General]
Is there any way where i can seperate my .templ files from my autogenerated _templ.go files? After a certain point as the project gets a bit bigger/ modular, the generated files all start cluttering my tree view of the folder structure.

If i can hide them that would also work but i guess the better way would be for me to provide a flag with a folder path to keep the generated files in, if i dont care about their package.

Any ideas/ ways?

**Comment by pharrisee** [comment]
If you're using VSCode it's possible to use the nesting functionality to 'hide' the generated files (while still allowing access if required).

Add the following into the VSCode config:

```json
    "explorer.fileNesting.enabled": true,
    "explorer.fileNesting.expand": false,
    "explorer.fileNesting.patterns": {
        "*.templ": "${basename}_templ.*",
    },
```
Example:

![image](https://github.com/user-attachments/assets/11665c7a-8891-4b27-b2ae-f42562b59490)

I also use this to hide what I call meta files in the root of the project, things like go.mod, go.sum, tailwind.config.js, env files etc. by using this rule:

```json
".meta": "go.*, *.yml, .git*, *.ts, *.md, *.json,*.js,Dockerfile*,.env,*.env,*.txt, *.service, *.http,*.sh,*.csv, *.toml,*.db, LICENSE"
```

This ends up giving me access but keeping the root looking clean:

![image](https://github.com/user-attachments/assets/a7a8c7b2-81a2-48b5-9b3d-9fe12a921406)

Simply create an empty file named `.meta` and presto all the meta files are hidden away.

Bear in mind that the nesting is only visual, no files were harmed or moved in the nesting process.

**Reply by joerdav** [reply]
I'd recommend the above as an approach. Moving generated files would have a few adverse effects, since Go's package system is very much file based!
